updateParamGroups.iLogicVb

```diff
@@ -65,81 +83,80 @@ Public Class updateParamGroup
 	
 		Select Case BeforeOrAfter
 			Case kBefore
-				'MsgBox("kBefore")
 
 			Case kAfter
-				'MsgBox("kAfter")
-				AfterParameterChange(oDoc, parameter)
+				NewName = parameter.Name
+				EditParameterGroup(oDoc)
 
 			Case Else
 				Exit Sub
 
 		End Select
 	End Sub
 
-	Private Sub BeforeParameterChange(oDoc As Inventor.Document, parameter As Inventor.Parameter)
-		'run before the parameter changes, move/delete changed parameters!
+	Private Function HasGroupChanged() As Boolean
+
+		If ParseGroupName(initialName) <> ParseGroupName(NewName) Then
+			Return true
+		Else
+			Return false
+		End If
 
-		initialName = parameter.Name
-		initialGroupName = parseGroupName(initialName)
+	End Function
+
+	Private Sub CleanupParameterGroup(oDoc As Inventor.Document)
+
+		if IsParameterGroupFormat(initialName) = false then
+			exit sub
+		end if
+
+		initialGroupName = ParseGroupName(initialName)
 		GetParamGroups(oDoc)
 
-		If StringContainsGroupName(initialName) = false Then
-			'msgbox("Exiting, StringContainsGroupName returned false")
+		If IsValidGroupName(initialName) = false Then
 			Exit Sub
 		End If
 
-		If DoesGroupNameExist(initialGroupName) = false Then
-			'msgbox("Exiting, DoesGroupNameExist returned false")
-			Exit Sub
-		End If
+		initialGroup = GetGroupByName(initialGroupName)
 
-		If DoesGroupContainParameter(initialName) = false Then
-			'msgbox("Exiting, DoesGroupContainParameter returned false")
+		If DoesGroupContainParameter(NewName, initialGroup) = false Then
 			Exit Sub
 		End If
 
-		initialGroup = GetGroupByName(initialGroupName)
-		RemoveParameterFromGroup(initialName, initialGroup)
+		RemoveParameterFromGroup(NewName, initialGroup)
 			
 		If IsGroupEmpty(initialGroup) Then
 			DeleteGroup(initialGroup)
 		End If
 
 	End Sub
 
-	Private Sub AfterParameterChange(oDoc As Inventor.Document, parameter As Inventor.Parameter)
-		'run after the parameter changes, create parameter groups if needed!
-		NewName = parameter.Name
-		NewGroupName = parseGroupName(NewName)
+	Private Sub EditParameterGroup(oDoc As Inventor.Document)
+
+		if IsParameterGroupFormat(NewName) = false then
+			exit sub
+		end if
+
+		NewGroupName = ParseGroupName(NewName)
 		GetParamGroups(oDoc)
 	
-		If StringContainsGroupName(NewName) = false Then
-''			msgbox("StringContainsGroupName = false, exiting...")
+		If IsValidGroupName(NewName) = false Then
 			Exit Sub
 		End If
 
-		If DoesGroupNameExist(NewGroupName) = false Then
-''			msgbox("DoesGroupNameExist = false, exiting...")
-			Exit Sub
-		End If			
-
 		If DoesParameterExist(NewGroupName) = true Then
-''			msgbox("DoesParameterExist(NewGroupName)= true, exiting...")
 			Exit Sub
 		End If
 
 		If DoesGroupExist(NewGroupName) = false Then
-''			msgbox("DoesGroupExist = false, addingGroup...")
 			AddGroup(NewGroupName)
 		End If
```

Comment:
I think the old way for this was better, with the try-catch, because it didn't need to loop